src/controllers/athlete-controller.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const async = require('async')
+const boom = require('boom')
+const uuid = require('uuid/v1')
+const util = require('../util')
+const Athlete = require('../models/athlete')
+
+exports.getAthletes = (req, res, next) => {
+  Athlete.find()
+    .sort([['lastName', 'ascending'], ['firstName', 'ascending']])
+    .exec(function (err, list) {
+      if (err) { return next(err) }
+      res.status(200).json(list.map(doc => util.fromMongoDb(doc)))
+    })
+}
+
+exports.getAthlete = (req, res, next) => {
+  // todo get GHIN index from ghin.com
+  async.parallel({
+    athlete: (callback) => {
+      Athlete.findById(req.params.id)
+        .exec(callback)
+    }
+  }, function (err, results) {
+    if (err) {
+      return next(err)
+    } else if (results.athlete == null) {
+      return next(boom.notFound())
+    }
+    res.status(200).json(util.fromMongoDb(results.athlete))
+  })
+}
+
+exports.createAthlete = (req, res, next) => {
+  // todo index ghinNumber to avoid collection scan
+  Athlete.findOne({ ghinNumber: req.body.ghinNumber })
+    .exec((err, athlete) => {
+      if (err) {
+        return next(err)
+      } else if (athlete && athlete.ghinNumber && athlete.ghinNumber === req.body.ghinNumber) {
+        return next(boom.badRequest('Athlete with this GHIN number already exists'))
+      }
+      const now = Date.now()
+      const newAthlete = new Athlete(Object.assign({ _id: uuid(), createdAt: now, updatedAt: now }, req.body))
+      newAthlete.save((err, result) => {
+        if (err) {
+          return next(err)
+        } else {
+          res.status(201).json(util.fromMongoDb(result))
+        }
+      })
+    })
+}
+
+exports.updateAthlete = (req, res, next) => {
+  Athlete.findById(req.params.id)
```

Comment:
You've added Promises to mongoose before (`mongoose.Promise = global.Promise`) so it would be a good idea to use them. Code would be much simpler.

Reply:
Agree. Will make the change.